### ISSUE TITLE
Add Intel TBB

### ISSUE THREAD

**Issue by wds15** [stan-dev/cmdstan#709]
#### Summary:
Make the Intel TBB available.

#### Description:
This is makes the use of the TBB available in Stan.

#### Additional Information:
Provide any additional information here.

#### Current Version:
v2.19.1

**Comment by abdmoussa** [comment]
Hello every body, i am new here 
I am trying to use intel TBB toolkit for testing some simple code, but when compiling with c++11 i get error message : 

```
Undefined symbols for architecture x86_64:
  "tbb::interface5::internal::task_base::destroy(tbb::task&)", referenced from:
      tbb::flow::interface10::graph::~graph() in TBB-7fdcbb.o
  "tbb::interface7::internal::task_arena_base::internal_terminate()", referenced from:
      tbb::interface7::task_arena::terminate() in TBB-7fdcbb.o
  "tbb::interface7::internal::task_arena_base::internal_initialize()", referenced from:
      tbb::interface7::task_arena::initialize() in TBB-7fdcbb.o
  "tbb::task_group_context::reset()", referenced from:
      tbb::flow::interface10::graph::wait_for_all() in TBB-7fdcbb.o
  "tbb::task_group_context::~task_group_context()", referenced from:
      tbb::flow::interface10::graph::~graph() in TBB-7fdcbb.o
 "tbb::interface7::internal::task_arena_base::internal_execute(tbb::interface7::internal::delegate_base&) const", referenced from:
      void tbb::interface7::task_arena::execute_impl<void, tbb::flow::interface10::graph::wait_functor const>(tbb::flow::interface10::graph::wait_functor const&) in TBB-7fdcbb.o
  "tbb::task_group_context::is_group_execution_cancelled() const", referenced from:
      tbb::flow::interface10::graph::wait_for_all() in TBB-7fdcbb.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Some body can help. 
Thanks in advance

**Comment by wds15** [comment]
@abdmoussa I am not sure what the issue is here... but so far we haven't supported TBB. This has changed since very recently. So you should be able to grab the latest develop cmdstan and a `make build` should give you a build Intel TBB library. Please open an issue if you have trouble.

Closing this issue which was addressed in PR #710
